### PR TITLE
Automatically add principals as web users from allowlist file

### PR DESCRIPTION
**Pull request by chiayi**
Add all the users/groups specified in the `allowlist` file as web app users for IAP.

### CODE REVIEW

File: jupyter-on-gke/jupyterhub.tf [hunk 1]
```diff
@@ -41,6 +41,14 @@ data "google_compute_backend_service" "jupyter-ingress" {
   project = var.project_id
 }
 
+resource "google_iap_web_backend_service_iam_binding" "binding" {
+  count = var.add_auth && data.google_compute_backend_service.jupyter-ingress.generated_id != null ? 1 : 0
```
Comment (imreddy13) [inline jupyter-on-gke/jupyterhub.tf:45]: data.google_compute_backend_service.jupyter-ingress.generated_id != null --> why do we need this?

File: jupyter-on-gke/variables.tf [hunk 2]
```diff
@@ -70,4 +70,10 @@ variable "url_domain_name" {
   type = string
   description = "Name of the domain provided by the user. This var will only be used if url_domain_addr is not empty"
   default = ""
+}
+
+variable "support_email" {
+  type = string
+  description = "Email displayed on the OAuth consent screen."
```
Comment (imreddy13) [inline jupyter-on-gke/variables.tf:77]: Maybe we can add a Pantheon link to where this email can be found?

File: jupyter-on-gke/jupyterhub.tf (hunk 1 above)
Comment (imreddy13) [reply jupyter-on-gke/jupyterhub.tf:45]: What happens if the backend service doesn't exist yet? Does the terraform apply fail?

Comment (chiayi) [reply jupyter-on-gke/jupyterhub.tf:45]: Yup, it gives an 404 error in terraform saying that the service doesn't exist.

File: jupyter-on-gke/variables.tf (hunk 2 above)
Comment (chiayi) [reply jupyter-on-gke/variables.tf:77]: Oh wait, this is part of the automatically creating clients, required field for creating a brand. Let me know if you would like me to create another PR for this. 

But without a brand, they won't see the email on the Pantheon page.

Comment (chiayi) [reply jupyter-on-gke/variables.tf:77]: I'll add a better description, not exactly telling you what it is.

File: jupyter-on-gke/variables.tf [hunk 3]
```diff
@@ -70,4 +70,10 @@ variable "url_domain_name" {
   type = string
   description = "Name of the domain provided by the user. This var will only be used if url_domain_addr is not empty"
   default = ""
+}
+
+variable "support_email" {
+  type = string
+  description = "Email for users to contact with questions about their consent"
```
Comment (imreddy13) [inline jupyter-on-gke/variables.tf:77]: Still think that the admin setting up this environment needs to know where to find this email. Can we provide either a link to the Pantheon oauth consent page where they can find it, or instructions for how to find it?

**Review by imreddy13 — APPROVED** [review]

File: jupyter-on-gke/variables.tf (hunk 3 above)
Comment (imreddy13) [reply jupyter-on-gke/variables.tf:77]: Ohh, I see - we just associate whatever email they provide with the brand we create? In which case, ignore this :)

**Pull request merged**